### PR TITLE
refactor: [Part 1] - Ongoing Work for legacy <Text> comp replacement.

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**
This PR replaces old usage of legacy <Text> component with the text component that standardizes on typography provided through theme via [@metamask/design-tokens](https://www.npmjs.com/package/@metamask/design-tokens) library. This will bring uniformity to all text elements on our app.

## **Manual testing steps**
- From the main screen, click on any asset available to go to Asset Detail.
- Make sure  you have no balance on the chosen asset [Specific for testing Warning Message Text].
- Review Button's Texts.

### **Before**
Screenshot 2023-10-20 at 14 54 13
Simulator Screen Shot - iPhone 12 Pro - 2023-10-23 at 12 17 49

### **After**
Screenshot 2023-10-20 at 15 07 52
Simulator Screen Shot - iPhone 12 Pro - 2023-10-23 at 12 27 21

## **Related issues**

This is **part** of ongoing tasks related to issue #6887

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx [hunk 1]
```diff
@@ -93,7 +95,11 @@ const AssetActionButton = ({
         {icon && (typeof icon === 'string' ? getIcon(icon) : icon)}
       </View>
       {label && (
-        <Text centered style={styles.buttonText} numberOfLines={1}>
+        <Text
+          variant={TextVariant.BodyMD}
```
Comment (owencraston) [inline app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:99]: The default value for variant is already `TextVariant.BodyMD` so I do not think you need to pass those values in here.

File: app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx [hunk 2]
```diff
@@ -93,7 +95,11 @@ const AssetActionButton = ({
...
+          style={styles.buttonText}
```
Comment (owencraston) [inline app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:100]: I looked through the styles that are passed here and I think we could clean it up a tad. 
This is how the button styles are defined

```
    buttonText: {
      marginTop: 8,
      marginHorizontal: 3,
      color: colors.primary.default,
      fontSize: 14,
    },
```

I think a better pattern for passing color to the Text component would be to use the `color` prop. That way we can migrate all the Text components to use "valid" [TextColor values](https://github.com/MetaMask/metamask-mobile/blob/664977cf0ce1b2e99cd4bfe10134f7193bdf5697/app/component-library/components/Texts/Text/Text.types.ts#L24-L25). In this case I would remove the color from buttonText and pass it in as `color={TextColor.Primary}`

Comment (owencraston) [reply app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:100]: I am wondering now is the `TextColor.Inverse` will be the same as `colors.primary.default`. @brianacnguyen is this valid?

File: app/components/UI/AssetOverview/AssetOverview.tsx [hunk 3]
```diff
@@ -124,12 +127,13 @@ const AssetOverview: React.FC<AssetOverviewProps> = ({
   const renderWarning = () => (
     <View style={styles.warningWrapper}>
       <TouchableOpacity
+        testID={TOKEN_OVERVIEW_BALANCE_WARNING}
         onPress={() => goToBrowserUrl(AppConstants.URLS.TOKEN_BALANCE)}
       >
-        <Text style={styles.warning}>
+        <Text variant={TextVariant.BodyMD} style={styles.warning}>
```
Comment (owencraston) [inline app/components/UI/AssetOverview/AssetOverview.tsx:133]: Again, I don't think the variant is needed here since it is using the default value.

Comment (owencraston) [reply app/components/UI/AssetOverview/AssetOverview.tsx:133]: I am looking at the warning styles and they are defined as follows
```
    warning: {
      ...typography.sBodyMD,
      borderRadius: 8,
      borderWidth: 1,
      borderColor: colors.warning.default,
      backgroundColor: colors.warning.muted,
      padding: 20,
    } as TextStyle,
    warningLinks: {
      ...typography.sBodyMD,
      color: colors.primary.default,
    } as TextStyle,
    ```
This indicates a few things to me. It seems that the style is setting the variant to use `sBodyMD`. In my opinion we should remove this from the styles and just set that as the variant through the props. I also find it very weird that we are using this Text component to render the warning banner and passing the colours in through styles but that is propably out of scope for this change.

File: app/components/UI/AssetOverview/AssetOverview.tsx [hunk 4]
```diff
@@ -124,12 +127,13 @@ const AssetOverview: React.FC<AssetOverviewProps> = ({
...
           {strings('asset_overview.were_unable')} {(asset as Asset).symbol}{' '}
           {strings('asset_overview.balance')}{' '}
-          <Text style={styles.warningLinks}>
+          <Text variant={TextVariant.BodyMD} style={styles.warningLinks}>
```
Comment (owencraston) [inline app/components/UI/AssetOverview/AssetOverview.tsx:136]: same comment as above ^.

File: app/components/UI/AssetOverview/Balance/Balance.tsx [hunk 5]
```diff
@@ -15,11 +17,15 @@ const Balance = ({ balance, fiatBalance }: BalanceProps) => {
   const { styles } = useStyles(styleSheet, {});
   return (
     <View style={styles.wrapper}>
-      <Text style={styles.text}>{strings('asset_overview.your_balance')}</Text>
+      <Text variant={TextVariant.BodyMD} style={styles.text}>
```
Comment (owencraston) [inline app/components/UI/AssetOverview/Balance/Balance.tsx:20]: text styles are defined as follows

```
    text: {
      ...typography.sBodySM,
      marginVertical: 0,
    } as TextStyle,
```

Given this lets remove that style from the `Balance.styles.ts` file  and pass in `sBodySM` as the variant so we don't override that variant.

File: app/components/UI/AssetOverview/Balance/Balance.tsx [hunk 6]
```diff
@@ -15,11 +17,15 @@ const Balance = ({ balance, fiatBalance }: BalanceProps) => {
...
+        {strings('asset_overview.your_balance')}
+      </Text>
       <Title style={styles.fiatBalance}>
         {fiatBalance || strings('asset_overview.unable_to_load_balance')}
       </Title>
-      <Text style={styles.text}>{balance}</Text>
+      <Text variant={TextVariant.BodyMD} style={styles.text}>
```
Comment (owencraston) [inline app/components/UI/AssetOverview/Balance/Balance.tsx:26]: same as above.

File: app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx (hunk 1 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:99]: Got u. Fixed.

File: app/components/UI/AssetOverview/AssetOverview.tsx (hunk 4 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/AssetOverview.tsx:136]: Fixed already.

File: app/components/UI/AssetOverview/Balance/Balance.tsx (hunk 6 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/Balance/Balance.tsx:26]: Fixed already.

File: app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx (hunk 2 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/AssetActionButton/AssetActionButton.tsx:100]: Did some clean up already removing props that are now present on the `TextVariant `. Just for the sake of registering here. Checking `design-tokens` repository I see that `TextColor.Inverse` is set to `#FCFCFC` and `colors.primary.default = #0376C9` thus not being the best replacement. Based on the following from `app/component-library/components/Texts/Text/Text.styles.ts` the best replacement in this case would be `TextColor.Primary`.

```
    case TextColor.Primary:
      textColor = theme.colors.primary.default;
      break;
```

File: app/components/UI/AssetOverview/AssetOverview.tsx (hunk 3 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/AssetOverview.tsx]: Removed unnecessary style attrs and added color as direct prop.

File: app/components/UI/AssetOverview/Balance/Balance.tsx (hunk 5 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/Balance/Balance.tsx]: Thanks, but in this case `sBodySM` is the default variant thus there's no need to pass it.

**Review by owencraston — DISMISSED** [review]

File: app/components/UI/AssetOverview/Balance/Balance.tsx [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { View } from 'react-native';
 import { strings } from '../../../../../locales/i18n';
-import Text from '../../../Base/Text';
+import Text from '../../../../component-library/components/Texts/Text';
 import Title from '../../../Base/Title';
 import { useStyles } from '../../../../component-library/hooks';
 import styleSheet from './Balance.styles';
```
Comment (brianacnguyen) [inline app/components/UI/AssetOverview/Balance/Balance.tsx:7]: shouldnt the text variant be BodySM since the Text default is BodyMD?

**Review by brianacnguyen — CHANGES_REQUESTED** [review]

File: app/components/UI/AssetOverview/Balance/Balance.tsx (hunk 7 above)
Comment (Jonathansoufer) [reply app/components/UI/AssetOverview/Balance/Balance.tsx:7]: You're write. fixed.

**Review by owencraston — APPROVED** [review]

**Pull request closed without merging**